radicle-node/src/service/reactor.rs

```diff
@@ -31,12 +31,42 @@ pub enum Io {
 pub struct Fetch {
     /// Repo to fetch.
     pub rid: Id,
-    /// Namespaces to fetch.
-    pub namespaces: Namespaces,
+    /// Indicates whether the fetch request was initiated or is a response.
+    pub direction: FetchDirection,
     /// Remote peer we are interacting with.
     pub remote: NodeId,
-    /// Indicates whether the fetch request was initiated by us.
-    pub initiated: bool,
+}
+
+impl Fetch {
+    pub fn is_initiated(&self) -> bool {
+        self.direction.is_initiated()
+    }
+
+    pub fn initiated(&self) -> Option<&Namespaces> {
+        match &self.direction {
+            FetchDirection::Initiator { namespaces } => Some(namespaces),
+            FetchDirection::Responder => None,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum FetchDirection {
+    /// Client is initiating a fetch in order to receive the specified
+    /// `refspecs` determined by [`Namespaces`].
+    Initiator {
+        /// Namespaces to fetch.
+        namespaces: Namespaces,
+    },
+    /// Server is responding to a fetch request by uploading the
+    /// specified `refspecs` sent by the client.
+    Responder,
+}
+
+impl FetchDirection {
+    pub fn is_initiated(&self) -> bool {
```

Comment:
Nit: maybe `is_initiator` when it's on `FetchDirection`?

Reply:
And why not both then? :thinking:

Reply:
Yeah actually both would also work for me :D 

It's just that `Fetch::is_initiated` is easier to swallow because it's about the whole fetch object, which isn't an enum. But for the enum, it's nice to have it match the variant.

Reply:
Ya, I think I'll switch to `is_initiator` for both then :)